### PR TITLE
Add search box

### LINKED ISSUE #48: Add a Search Box
Follow the [designs here](https://www.figma.com/file/MYWmd1buvc2AMvUmFP9w42t5/Browser?node-id=3573%3A12049) to add a search box on the home page.

Search items should show up on each new key press after 2 letters.

### PR DESCRIPTION
**Pull request by iamonuwa**
Fixes #48

### CODE REVIEW

File: src/common/components/Search/SearchResultItem/SearchResultItem.jsx [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+      <div className={styles.imgWrapper}>
+        <ReactImageFallback
+          className={styles.image}
+          src={image}
+          fallbackImage={icon}
+          alt="App icon"
+        />
+      </div>
+      <div className={styles.column}>
+        <>
```
Comment (andytudhope) [inline src/common/components/Search/SearchResultItem/SearchResultItem.jsx]: Is this empty tag necessary?

**Review by andytudhope — COMMENTED** [review]
One comment on the code and 2 general comments:

1. The search function is not very clever. For instance, I have a DApp, "Local Test" - yet searching just for "Test" does not show anything. Any chance we could make the search match any word, rather than having to begin with exactly the same letters?

2. The search box does not show on Desktop. I think we should make this a separate bounty, as this is already a lot of work for the amount, and we'll need a desktop design for how to display the search results cc @errorists

**Comment by iamonuwa** [comment]
I'll update the logic to search everything. Currently its searching with .startsWith()

File: src/common/components/Search/SearchResultItem/SearchResultItem.jsx (hunk 1 above)
Comment (iamonuwa) [reply src/common/components/Search/SearchResultItem/SearchResultItem.jsx]: Removed

**Comment by errorists** [comment]
@andytudhope sure, [here's the design](https://www.figma.com/file/MYWmd1buvc2AMvUmFP9w42t5/Browser?node-id=3667%3A149)

The box expands on focus, shrinks on defocus. results are shown similar to any other list or mobile

**Comment by iamonuwa** [comment]
> @andytudhope sure, [here's the design](https://www.figma.com/file/MYWmd1buvc2AMvUmFP9w42t5/Browser?node-id=3667%3A149)
> 
> The box expands on focus, shrinks on defocus. results are shown similar to any other list or mobile

This design is different from [this](https://www.figma.com/file/MYWmd1buvc2AMvUmFP9w42t5/Browser?node-id=3573%3A12049)

**Comment by errorists** [comment]
yeah, it's floating on top similar to the hamburger menu. I've started out with something exactly similar [here](https://www.figma.com/file/MYWmd1buvc2AMvUmFP9w42t5/Browser?node-id=1881%3A2990) but it wouldn't perform as good. Apply the necessary css change for the input field in the desktop breakpoint.

**Comment by andytudhope** [comment]
Yeah @iamonuwa the design is different from when we started, so if you're able to do it as a part of this PR, I will add 100 DAI tip to the bounty.

**Comment by iamonuwa** [comment]
> Yeah @iamonuwa the design is different from when we started, so if you're able to do it as a part of this PR, I will add 100 DAI tip to the bounty.

Am on it now.

**Comment by iamonuwa** [comment]
> yeah, it's floating on top similar to the hamburger menu. I've started out with something exactly similar [here](https://www.figma.com/file/MYWmd1buvc2AMvUmFP9w42t5/Browser?node-id=1881%3A2990) but it wouldn't perform as good. Apply the necessary css change for the input field in the desktop breakpoint.

@errorists do I leave the mobile as is or use same as desktop in new design?

**Comment by errorists** [comment]
Leave mobile as is please 🙏

**Comment by iamonuwa** [comment]
> Leave mobile as is please

Alright, thanks

**Comment by andytudhope** [comment]
OK @iamonuwa we're nearly there, nice work! Two problems on the desktop stuff though:

1. On focus, it expands the box, as expected. I can then search, but if I type a word in and then defocus, it still disappears, which is unexpected. If there is text present, it shouldn't disappear.

2. Can you fix the alignment of the displayed results with the box, as in the design. This is what it currently looks like for me. You should not be able to see those corners.

Screenshot from 2019-09-13 11-09-10

**Comment by iamonuwa** [comment]
Thanks for your feedback. Am fixing those now.

**Comment by andytudhope** [comment]
Hey @iamonuwa - I'm still not quite happy with the how the search results appear. I did not mean that you should merely move the box up a bit to try and hide the corners - it needs to blend seamlessly in with the searchbox, as shown in the designs...

Screenshot from 2019-09-16 09-28-22

**Comment by iamonuwa** [comment]
@andytudhope I'll use css now to fix this

**Pull request merged**